src/components/AgreementSections/InfoPeriods.js

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  Button,
+  ButtonGroup,
+  Col,
+  FormattedUTCDate,
+  Headline,
+  KeyValue,
+  Layout,
+  NoValue,
+  Row,
+} from '@folio/stripes/components';
+
+const InfoPeriods = ({ periods }) => {
+  /*
+   * This could be handled by callback, but since we need to know for display whether next/prev/current exist
+   * it seems more prudent to just fetch all three at once and then do our logic at this stage
+   */
+
+  const displayPeriodOptions = periods?.reduce((map, period) => {
+    if (period.periodStatus) map[period.periodStatus] = period;
+    return map;
+  }, {}) || {};
+
+  /*
+   * Set up which period is currently selected. Heirachy is Current > Next > Previous
+   * The empty array as a second argument to useEffect should mean this code only runs once per component lifecycle
+   * to prevent infinite loops.
+   * This is equivalent to the old "componentDidMount"
+   */
+  const [selectedPeriod, setSelectedPeriod] = useState('current');
+
+  useEffect(() => {
+    if (!displayPeriodOptions.current) {
+      if (displayPeriodOptions.next) {
+        setSelectedPeriod('next');
+      } else {
+        setSelectedPeriod('previous');
```

Comment:
Can we refactor this to be
```
setSelectedPeriod(prevState => {
    if (prevState.current) {
      return displayPeriodOptions.next ? 'next' : 'previous';
    }
})

Reply:
No, because the "selectedPeriod" is simply a string, so prevState will not have a .current

Reply:
I can however make it: 
```
if (!displayPeriodOptions.current) {
      setSelectedPeriod(() => {
        return displayPeriodOptions.next ? 'next' : 'previous';
      })
    }
```

It needs the external if, if that is inside the setSelectedPeriod then when displayPeriodOptions does indeed contain `current`, the setSelectedPeriod returns undefined. I don't particularly want to have to return the value it's already set to if nothing needs to change, so I wrapped the whole setSelectedPeriod